secrethub_wrapper.go

```diff
@@ -116,18 +118,24 @@ func Resolve(ref *C.char, errMessage **C.char) *C.char {
 	return ref
 }
 
-/*
 // ResolveEnv takes a map of environment variables and replaces the values of those
 // which store references of secrets in SecretHub (`secrethub://<path>`) with the value
 // of the respective secret. The other entries in the map remain untouched.
 //export ResolveEnv
-func ResolveEnv(envVars map[string]string) map[string]string {
+func ResolveEnv(errMessage **C.char) *C.char {
+	envVars := os.Environ()
 	resolvedEnv := make(map[string]string, len(envVars))
-	for key, value := range envVars {
-		resolvedEnv[key] = string(Resolve(value))
+	for _, value := range envVars {
+		keyValue := strings.Split(value, "=")
+		resolvedEnv[keyValue[0]] = C.GoString(Resolve(keyValue[1], errMessage))
+	}
+	encoding, err := json.Marshal(resolvedEnv)
+	if err != nil {
+		*errMessage = C.CString(err.Error())
+		return nil
 	}
-	return resolvedEnv
-}*/
+	return C.CString(encoding)
+}
```

Comment:
This function should call the `ResolveEnv` method implemented on the Go client.

Reply:
Adjusted. Now the call to `secrethub-go` SDK is made.

Reply:
As you know, we decided to postpone adding this functionality to the go client. I'll revert these changes (sorry about that).